### ISSUE TITLE
Error: theta = 0 for a population

### ISSUE THREAD

**Issue by JasonFN** [simjoly/jml#1]
Hi,

I am currently trying to use the JML software but have encountered a problem when running it on terminal on a Mac computer. Would anyone be able to help with this?

I'll include my files here (species.trees, tpi.phy, and jml.tpi.ctl)

[jml.tpi.ctl.txt](https://github.com/simjoly/jml/files/2934703/jml.tpi.ctl.txt)
[tpi.phy.txt](https://github.com/simjoly/jml/files/2934704/tpi.phy.txt)
[species.trees.trees.txt](https://github.com/simjoly/jml/files/2934705/species.trees.trees.txt)

Many thanks in advance!
Francis

Here is the simulation log:

######################################

Reading control file... done.

 Reading tree file "species.trees"... done.

 *** Species trees file information ***
 Number of trees read: 2001
 Number of trees in burnin: 0
 Number of trees post-burnin: 2001
 Number of trees sampled for simulations: 2001
 Seed used: 1551852531
 Significance level: 0.1

 *** Locus-specific information ***
 Locus relative mutation rate: 1
 Locus heredity scalar: 1

 Performing simulation 1
Error: theta=0 for a population.
staff-210-247:Banksia_jml FrancisNge$

**Comment by ShuiyinLIU** [comment]
Hi Francis,

Have you fixed this error of JML? 
I also encountered the same problem when running JML on linux. I found all the population size (theta) in file Rep.dat (generated by JML) are 0. For example, "((((((((((((((HRmore #0,LOtrin #0):0.000475333437543 #0,((CEcast #0,LObume #0):.....".I tried a lot to fix this problem, but still not work. If you or someone else know how to fix, could you tell me?

sincerely,
shuiyin

**Comment by ShuiyinLIU** [comment]
Hi Francis,

I just found out what's wrong. The input file of *species.trees from the latest starBEAST2 (version 0.15.5) I used seems not compatible with JML (version 1.3.1). 

The *species.trees generated by starBEAST2, for instance, like this:
tree STATE_19595000 = ((((((((1[&dmv={0.0014810016026519232}]:6.626672124751781E-,154[&dmv={4.8662842684255513E-4}]:6.626672124751781E-5)..............
The *species.trees provided by JML for example, like this:
tree STATE_1180000 [&lnP=-4850.566978820145] = [&R] ((3[&dmv=0.00539197244247676]:0.002862394043860964,..........

As for information of the population size provide by *species.trees from starBEAST2, here's an extra pair of braces (e.g.[&dmv={0.0014810016026519232}] compared with example file of JML (e.g. [&dmv=0.00539197244247676]). I guess this would cause the JML can't calculate the population size (i.e. theta) for each species from *species.trees, so return such error.

When I removed all the braces (i.e. {}) in *species.trees, JML worked well now. Therefore, perhaps you modified your species trees file (i.e. species.trees.trees.txt) and make its format same with the example file (a slightly modified nexus format ) provided by JML, then maybe it will work. Hope this will be useful for you.

Shuiyin

**Comment by ShuiyinLIU** [comment]
Hi Francis,

I just found out what's wrong. The input file of *species.trees from the latest starBEAST2 (version 0.15.5) I used seems not compatible with JML (version 1.3.1). 

The *species.trees generated by starBEAST2, for instance, like this:
tree STATE_19595000 = ((((((((1[&dmv={0.0014810016026519232}]:6.626672124751781E-,154[&dmv={4.8662842684255513E-4}]:6.626672124751781E-5)..............
The *species.trees provided by JML for example, like this:
tree STATE_1180000 [&lnP=-4850.566978820145] = [&R] ((3[&dmv=0.00539197244247676]:0.002862394043860964,..........

As for information of the population size provide by *species.trees from starBEAST2, here's an extra pair of braces (e.g.[&dmv={0.0014810016026519232}] compared with example file of JML (e.g. [&dmv=0.00539197244247676]). I guess this would cause the JML can't calculate the population size (i.e. theta) for each species from *species.trees, so return such error.

When I removed all the braces (i.e. {}) in *species.trees, JML worked well now. Therefore, perhaps you modified your species trees file (i.e. species.trees.trees.txt) and make its format same with the example file (a slightly modified nexus format ) provided by JML, then maybe it will work. Hope this will be useful for you.

Shuiyin
